### PR TITLE
feat: mvi search and fetch vectors

### PR DESCRIPTION
**Pull request by malandis**
Adds `searchAndFetchVectors` method to both web and Node.js clients, including:
- method definitions
- response types (+`SearchAndFetchVectorsHit`)
- integration tests

For the integration tests I parameterized the existing search tests to work on both `search` and `searchAndFetchVectors`.

### CODE REVIEW

File: packages/core/src/internal/clients/vector/AbstractVectorIndexClient.ts [hunk 1]
```diff
@@ -133,6 +135,34 @@ export abstract class AbstractVectorIndexClient
     return await this.dataClient.search(indexName, queryVector, options);
   }
 
+  /**
+   * Searches for the most similar vectors to the query vector in the index.
+   *
+   * Ranks the vectors according to the similarity metric specified when the
+   * index was created.
+   * Also retursn the vectors associated with each result.
```
Comment (nand4011) [inline packages/core/src/internal/clients/vector/AbstractVectorIndexClient.ts:143]: Minor typo.

Comment (malandis) [inline packages/core/src/internal/clients/vector/AbstractVectorIndexClient.ts:143]:
```suggestion
   * Also returns the vectors associated with each result.
```

Comment (malandis) [reply packages/core/src/internal/clients/vector/AbstractVectorIndexClient.ts]: Good catch. Fixed in [297d469](https://github.com/momentohq/client-sdk-javascript/pull/1034/commits/297d469e0351c00ac425082d52a40dfdace5b447)

**Review by nand4011 — APPROVED** [review]
LGTM. The python sdk keeps SearchHit separate from the search vector. Do we want to change that to combine them like you do here?

**Comment by malandis** [comment]
> LGTM. The python sdk keeps SearchHit separate from the search vector. Do we want to change that to combine them like you do here?

I'm not sure what you mean?

**Comment by nand4011** [comment]
Ignore me. That was a poor reading of the python search hit.

**Pull request merged**